src/hardhat.rs

```diff
@@ -0,0 +1,101 @@
+use std::sync::{Arc, RwLock};
+
+use crate::{fork::ForkSource, node::InMemoryNodeInner};
+use jsonrpc_core::{BoxFuture, Result};
+use jsonrpc_derive::rpc;
+use zksync_basic_types::{AccountTreeId, Address, U256};
+use zksync_core::api_server::web3::backend_jsonrpc::error::into_jsrpc_error;
+use zksync_types::{utils::storage_key_for_standard_token_balance, L2_ETH_TOKEN_ADDRESS};
+use zksync_utils::u256_to_h256;
+use zksync_web3_decl::error::Web3Error;
+
+/// Implementation of HardhatNamespaceImpl
+pub struct HardhatNamespaceImpl<S> {
+    node: Arc<RwLock<InMemoryNodeInner<S>>>,
+}
+
+impl<S> HardhatNamespaceImpl<S> {
+    /// Creates a new `Hardhat` instance with the given `node`.
+    pub fn new(node: Arc<RwLock<InMemoryNodeInner<S>>>) -> Self {
+        Self { node }
+    }
+}
+
+#[rpc]
+pub trait HardhatNamespaceT {
+    #[rpc(name = "hardhat_setBalance")]
```

Comment:
For verification evidence, can you include a screenshot of the endpoint working via `curl` or the `test_endpoints.http` file (if you have VSCode)? I just want to get in the habit of verifying full end-to-end for PR evidence of new features instead of only unit tests.

## Example
Account balance before:
```bash
curl --request POST \
  --url http://localhost:8011/ \
  --header 'content-type: application/json' \
  --data '{"jsonrpc": "2.0","id": "1","method": "eth_getBalance","params": ["0xa61464658AfeAf65CccaaFD3a512b69A83B77618", "latest"]}'
```


Update balance:
```bash
curl --request POST \
  --url http://localhost:8011/ \
  --header 'content-type: application/json' \
  --data '{"jsonrpc": "2.0","id": "1","method": "hardhat_setBalance","params": ["0xa61464658AfeAf65CccaaFD3a512b69A83B77618", 1337]}'
```

Account balance after:
```bash
curl --request POST \
  --url http://localhost:8011/ \
  --header 'content-type: application/json' \
  --data '{"jsonrpc": "2.0","id": "1","method": "eth_getBalance","params": ["0xa61464658AfeAf65CccaaFD3a512b69A83B77618", "latest"]}'
```

Reply:
![image](https://github.com/matter-labs/era-test-node/assets/140627974/00639c62-9d18-4709-bb2d-890c59fa4780)

i changed your example slightly- it takes an hex-encoded U256 for balance, not plain number

added to `test_endpoints.http` too

Reply:
Awesome, thank you! I've moved the image to the PR description for easier visibility (when PR conversations are closed, they are hidden and difficult to find)